openvino_xai/methods/black_box/aise.py

```diff
@@ -151,74 +265,209 @@ def _preset_parameters(
+            iterations = 150
+            divs = np.linspace(8, 1, 5)
+        else:
+            raise ValueError(f"Preset {preset} is not supported.")
+
+        if num_iterations_per_kernel is None:
+            num_iterations_per_kernel = iterations
+        if divisors is None:
+            divisors = divs
+        return num_iterations_per_kernel, divisors
+
+    def _process_box(self, padding_coef: float = 0.5) -> None:
+        target_box_scaled = [
+            self.target_box[0] / self.input_size[1],  # x1
+            self.target_box[1] / self.input_size[0],  # y1
+            self.target_box[2] / self.input_size[1],  # x2
+            self.target_box[3] / self.input_size[0],  # y2
+        ]
+        box_width = target_box_scaled[2] - target_box_scaled[0]
+        box_height = target_box_scaled[3] - target_box_scaled[1]
+        self._min_box_size = min(box_width, box_height)
+        self.kernel_widths = [self._min_box_size / div for div in self.divisors]
+
+        x_from = max(target_box_scaled[0] - box_width * padding_coef, 0.0)
+        x_to = min(target_box_scaled[2] + box_width * padding_coef, 1.0)
+        y_from = max(target_box_scaled[1] - box_height * padding_coef, 0.0)
+        y_to = min(target_box_scaled[3] + box_height * padding_coef, 1.0)
+        self.bounds = Bounds([x_from, y_from], [x_to, y_to])
+
+    def _get_loss(self, data_perturbed: np.array) -> float:
+        """Get loss for perturbed input."""
+        forward_output = self.model_forward(data_perturbed, preprocess=False)
+        boxes, pred_scores, labels = self.postprocess_fn(forward_output)
+
+        loss = 0
+        for box, pred_score, label in zip(boxes, pred_scores, labels):
+            if label == self.target_label:
+                loss = max(loss, self._iou(self.target_box, box) * pred_score)
+        return loss
+
+    @staticmethod
+    def _iou(box1: np.ndarray | List[float], box2: np.ndarray | List[float]) -> float:
+        box1 = np.asarray(box1)
+        box2 = np.asarray(box2)
+        tl = np.vstack([box1[:2], box2[:2]]).max(axis=0)
+        br = np.vstack([box1[2:], box2[2:]]).min(axis=0)
+        intersection = np.prod(br - tl) * np.all(tl < br).astype(float)
+        area1 = np.prod(box1[2:] - box1[:2])
+        area2 = np.prod(box2[2:] - box2[:2])
+        return intersection / (area1 + area2 - intersection)
+
+    def _update_metadata(
+        self,
+        boxes: np.ndarray | List,
+        scores: np.ndarray | List[float],
+        labels: np.ndarray | List[int],
+        target: int,
+        original_size: Tuple[int, int],
+    ) -> None:
+        x1, y1, x2, y2 = boxes[target]
+        width_scale = original_size[1] / self.input_size[1]
+        height_scale = original_size[0] / self.input_size[0]
+        x1, x2 = x1 * width_scale, x2 * width_scale
+        y1, y2 = y1 * height_scale, y2 * height_scale
+        self.metadata[Task.DETECTION][target] = [x1, y1, x2, y2], scores[target], labels[target]
```

Comment:
Can we discuss the `metadata` naming? In my understanding, metadata is usually something we don't change, something provided, like for an image the metadata will be `original size` or something like that. Here we calculate this `metadata` that feels a bit non-intuitive. 
I'd suggest naming it as `self.predictions` or `self.predicted_bboxes` and making it `None` by default. 

One more thing: if we introduce `metadata` only for AISE detection, do we need the dict with the task as the key? Or is it a draft for future improvements?

Reply:
> metadata is usually something we don't change, something provided

From the user perspective, this will be exactly like you described.

Reply:
> I'd suggest naming it as self.predictions or self.predicted_bboxes

I propose to keep it as metadata instead of e.g. prediction, because we might want to extend it further with e.g. some per-map metric value, etc.

Reply:
I propose to introduce metadata for classification as well. But for classification probably it will be optional. I will see in the future PRs.